src/lib/Libifl/pbs_delstatfree.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 1994-2020 Altair Engineering, Inc.
+ * For more information, contact Altair at www.altair.com.
+ *
+ * This file is part of both the OpenPBS software ("OpenPBS")
+ * and the PBS Professional ("PBS Pro") software.
+ *
+ * Open Source License Information:
+ *
+ * OpenPBS is free software. You can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Commercial License Information:
+ *
+ * PBS Pro is commercially licensed software that shares a common core with
+ * the OpenPBS software.  For a copy of the commercial license terms and
+ * conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+ * Altair Legal Department.
+ *
+ * Altair's dual-license business model allows companies, individuals, and
+ * organizations to create proprietary derivative works of OpenPBS and
+ * distribute them - whether embedded or bundled with other software -
+ * under a commercial license agreement.
+ *
+ * Use of Altair's trademarks, including but not limited to "PBS™",
+ * "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+ * subject to Altair's trademark licensing policies.
+ */
+
+/**
+ * @file	pbs_statfree.c
+ * @brief
+ * The function that deallocates a "batch_status" structure
+ */
+
+#include <pbs_config.h>   /* the master config generated by configure */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include "libpbs.h"
+
+
+/**
+ * @brief
+ *	-The function that deallocates a "batch_deljob_status" structure
+ *
+ * @param[in] bsp - pointer to batch request.
+ *
+ * @return	Void
+ *
+ */
+void
+__pbs_delstatfree(struct batch_deljob_status *bsp)
```

Comment:
Do we really need separate file for this small func? Can't we put this func in int_rdrpy.c itself?

Reply:
That's what the convention we are following in PBS, refer __pbs_statfree(), I thought we should follow the same.

Reply:
I understand that in past we have added in separate files which I don't like at all for small funcs, but for this I think we should put this small func in int_rdrpy.c and in future if needed then we put all pbs_*free in one file rather than in individual file